### PR TITLE
MSMP resources are cleaned up in cluster mode.

### PR DESCRIPTION
**Pull request by ravindraranwala**

### CODE REVIEW

**Comment by ravindraranwala** [comment]
code was cleaned up https://github.com/wso2/wso2-synapse/pull/209

File: modules/core/src/main/java/org/apache/synapse/message/processor/MessageProcessorCleanupService.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.apache.synapse.message.processor;
+
+import java.io.Serializable;
+import java.util.concurrent.Callable;
+
+public interface MessageProcessorCleanupService extends Callable<Void>, Serializable {
+    void setName(final String name);
+
```
Comment (shafreenAnfar) [inline modules/core/src/main/java/org/apache/synapse/message/processor/MessageProcessorCleanupService.java]: Remove this additional space.

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java [hunk 2]
```diff
@@ -108,7 +115,8 @@ public void init(SynapseEnvironment se) {
 
 	}
```
Comment (shafreenAnfar) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: It seems like we have to fix indentation.

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java [hunk 3]
```diff
@@ -388,18 +415,49 @@ private BlockingMsgSender initMessageSender(Map<String, Object> params) {
...
+            for (MessageConsumer messageConsumer : messageConsumers) {
+                messageConsumer.cleanup();
+            }
+        }
+    }
+
+    private Callable<Void> getMessageProcessorCleanupTask() {
+        MessageProcessorCleanupService cleanupTask = null;
+        if (logger.isDebugEnabled()) {
+            logger.debug("Trying to fetch InboundRequestProcessor from classpath.. ");
```
Comment (shafreenAnfar) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: Comment should have either one "." or three

**Comment by ravindraranwala** [comment]
These issues were solved by the pull request [1] 

[1] https://github.com/wso2/wso2-synapse/pull/242

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java [hunk 4]
```diff
@@ -32,6 +36,8 @@
 import org.apache.synapse.task.TaskManager;
 import org.apache.synapse.task.TaskManagerObserver;
 
+import sun.misc.Service;
```
Comment (prabathabey) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: Avoid using low level private APIs such ash sun.misc.Service, which can lead to interoperability issues if the same code is run on a JDK introduced by a different vendor. The same API was moved into a public API in all new JDK versions, namely, ServiceLoader, which can effectively be used as an alternative.

Comment (ravindraranwala) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: This issue was fixed in https://github.com/wso2/wso2-synapse/pull/242

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java (hunk 2 above)
Comment (ravindraranwala) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: This issue was fixed in https://github.com/wso2/wso2-synapse/pull/253

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java (hunk 3 above)
Comment (ravindraranwala) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: This issue was fixed in https://github.com/wso2/wso2-synapse/pull/253

File: modules/core/src/main/java/org/apache/synapse/message/processor/MessageProcessorCleanupService.java (hunk 1 above)
Comment (ravindraranwala) [inline modules/core/src/main/java/org/apache/synapse/message/processor/MessageProcessorCleanupService.java]: This issue was fixed in https://github.com/wso2/wso2-synapse/pull/253

**Pull request merged**